assets/js/components/dashboard-splash/dashboard-splash-app.js

```diff
@@ -16,135 +16,24 @@
  * limitations under the License.
  */
 
-/**
- * WordPress dependencies
- */
-import { Component, Fragment } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
-
 /**
  * Internal dependencies
  */
-import DashboardSplashMain from './dashboard-splash-main';
-import DashboardSplashNotifications from './dashboard-splash-notifications';
-import { trackEvent } from '../../util';
 import '../publisher-wins';
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_SITE } from '../../googlesitekit/datastore/site/constants';
 import SetupUsingProxy from '../setup/setup-proxy';
-import SetupUsingGCP from '../setup';
-import SetupWrapper from '../setup/setup-wrapper';
-
-const AUTHENTICATION = 1;
-const SETUP = 2;
-
-class DashboardSplashApp extends Component {
-	constructor( props ) {
-		super( props );
-
-		const { connectURL } = global._googlesitekitLegacyData.admin;
-
-		const {
-			showModuleSetupWizard,
-			isAuthenticated,
-			isVerified,
-			hasSearchConsoleProperty,
-		} = global._googlesitekitLegacyData.setup;
-
-		const {
-			canAuthenticate,
-			canSetup,
-			canViewDashboard,
-			canPublishPosts,
-		} = global._googlesitekitLegacyData.permissions;
-
-		this.state = {
-			showAuthenticationSetupWizard: canSetup && ( ! isAuthenticated || ! isVerified || ! hasSearchConsoleProperty ),
-			showModuleSetupWizard,
-			canViewDashboard,
-			canPublishPosts,
-			buttonMode: 0,
-			connectURL,
-		};
-
-		if ( canAuthenticate && ! isAuthenticated ) {
-			this.state.buttonMode = AUTHENTICATION;
-		}
-		if ( canSetup && ( ! isAuthenticated || ! isVerified || ! hasSearchConsoleProperty ) ) {
-			this.state.buttonMode = SETUP;
-		}
-
-		this.openAuthenticationSetupWizard = this.openAuthenticationSetupWizard.bind( this );
-		this.gotoConnectURL = this.gotoConnectURL.bind( this );
-	}
-
-	async openAuthenticationSetupWizard() {
-		await trackEvent( 'plugin_setup', 'setup_sitekit' );
-
-		this.setState( {
-			showAuthenticationSetupWizard: true,
-		} );
-	}
-
-	async gotoConnectURL() {
-		this.setState( {
-			showAuthenticationInstructionsWizard: false,
-			showAuthenticationSetupWizard: false,
-		} );
+import LegacyDashboardSplashApp from './LegacyDashboardSplashApp';
+const { useSelect } = Data;
 
-		await trackEvent( 'plugin_setup', 'connect_account' );
+export default function DashboardSplashApp() {
```

Comment:
Let's also rename the file to match the component name like the other files already have been.